src/client-handler.ts

```diff
@@ -0,0 +1,46 @@
+import * as tmi from "tmi.js";
+import { IChatCommand } from "./chat-command.interface";
+import { GreetingCommand } from "./commands/greeting.model";
+
+const availableCommands: IChatCommand[] = [new GreetingCommand()];
+
+export function processClientMessage(target: string, sender: tmi.Userstate, msg: string) {
+    if (msg.indexOf("!") < 0) {
+        console.log(`* message dropped: no command - message: ${msg}, sender: ${sender.username}`);
+        return;
+    }
+
+    console.log(`* command received: no command - message: ${msg}, sender: ${sender.username}`);
+
+    const commandsWithReceiver = msg.match(/!\w+(\s*@[\w|\d]*)*/g);
```

Comment:
just for understanding : which regexp scheme do we follow here ? maybe a short explanation / comment what \w, \s and \d are could be nice  (personal preference) :)

Reply:
I don't understand your concern here right now tbh. There is a predefined character class that is the same no matter what language. This includes that \d stands for a digit, ... (see https://de.wikipedia.org/wiki/Regul%C3%A4rer_Ausdruck#Vordefinierte_Zeichenklassen ) 🤔

Reply:
Sorry for the confusion. Seems this was just a little knowledge gap of mine. (basically I forgot the syntax of those character classes :joy: , so I see no problem with the code here. )